simulator-spring-boot/src/test/java/org/citrusframework/simulator/service/impl/DefaultScenarioExecutorServiceImplIT.java

```diff
@@ -0,0 +1,75 @@
+package org.citrusframework.simulator.service.impl;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.PessimisticLockException;
+import jakarta.persistence.RollbackException;
+import org.citrusframework.simulator.IntegrationTest;
+import org.citrusframework.simulator.scenario.SimulatorScenario;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.TestPropertySource;
+
+import java.util.concurrent.ExecutorService;
+
+import static java.util.Collections.emptyList;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.util.ReflectionTestUtils.setField;
+
+@IntegrationTest
+@ExtendWith({MockitoExtension.class})
+@TestPropertySource({"classpath:META-INF/citrus-simulator.properties"})
+@TestPropertySource(properties = {"citrus.simulator.pessimistic-lock-timeout=0"})
+class DefaultScenarioExecutorServiceImplIT {
+
+    @Mock
+    private ExecutorService executorServiceMock;
+
+    @Autowired
+    private EntityManagerFactory entityManagerFactory;
+
+    @Autowired
+    private DefaultScenarioExecutorServiceImpl fixture;
+
+    @BeforeEach
+    void beforeEachSetup() {
+        setField(fixture, "executorService", executorServiceMock, ExecutorService.class);
+    }
+
+    @Test
+    void transactionBoundariesAndLockOnAsyncScenarioExecution() {
+        Long scenarioExecutionId = fixture.run(mock(SimulatorScenario.class), "ScenarioName", emptyList());
+        assertNotNull(scenarioExecutionId);
+
+        // At this point, the ScenarioExecution should be persisted, but locked!
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            entityManager.getTransaction().begin();
+            // assertThrows(PessimisticLockException.class, () -> entityManager.createNativeQuery("SELECT count(execution_id) FROM scenario_execution WHERE execution_id=:" + ScenarioExecution_.EXECUTION_ID).setParameter(ScenarioExecution_.EXECUTION_ID, scenarioExecutionId).getFirstResult());
```

Comment:
@tschlat I fail to prove the desired behavior.. maybe you have more luck, but the entity seems not at all be available for selection, despite `EntityManager#flush`. deletion of the table fails however, so the lock seems to be in place. but it looks like it somehow "hides" the underlying entity entirely.

the jpa specification just says that the provider must "hinder the consumer from selection until the modifying transaction has been committed".. in my understanding, this does not imply that the consumer has to be blocked, but returning `null` is a completely viable alternative.

I am no longer looking into this, for the weekend. feel free to play with it!